webapp/TargetedMS/js/QCPlotHelperBase.js

```diff
@@ -130,15 +130,18 @@ Ext4.define("LABKEY.targetedms.QCPlotHelperBase", {
 
         LABKEY.Ajax.request({
             url: LABKEY.ActionURL.buildURL('targetedms', 'GetQCPlotsData.api'),
-            success: this.processPlotData,
+            success: function(response) {
+                this.lastParsedResponse = JSON.parse(response.responseText);
+                this.processPlotData();
+            },
             failure: LABKEY.Utils.getCallbackWrapper(this.failureHandler),
             scope: this,
             jsonData: plotsConfig
         });
     },
 
-    processPlotData: function(response) {
-        var parsed = JSON.parse(response.responseText);
+    processPlotData: function() {
+        var parsed = this.lastParsedResponse;
```

Comment:
I can't remember the flow, but would this be always present in processPlotData? Would it make sense to check if (this.lastParsedResponse)?

Reply:
Good suggestion. I tested and if you click to change the plotting options before the AJAX call is complete it won't be initialized yet. I added a check